### ISSUE TITLE
Media controls

### ISSUE THREAD

**Issue by bretanac93** [muammar/mkchromecast#207]
Hello, is there any exposed way to control the multimedia (video position, volume, play and pause)?
Some sort of endpoint would be awesome, then I can develop my own app to interact with this awesome project in a better way.

**Comment by muammar** [comment]
Hi, that can be worked out. What kind of app are you thinking about? I started this https://github.com/muammar/mkchromecast/tree/ui but have not had the time to advance more. I ask you because I would be willing to join forces to do a graphical controller.

**Comment by bretanac93** [comment]
I'm thinking about a remote controller, either as a website or a mobile app (react-native perhaps 🤔)
I'm willing to collaborate as I fell in love w/ this project :)

**Comment by muammar** [comment]
That would be great to have a graphical interface that others could use. I was starting doing something with PyQt but probably `react` would look more modern and beautiful. So yes, count me in. I would like to work on something like that. I probably should change the way this repo operates to make it easier for others to push changes.

**Comment by muammar** [comment]
>Yeah, could be a good idea also to expose the same exact functionalities you have included in the CLI (p - pause, r - resume, u - volume up, d - volume down) as an endpoint, anyways you are starting a web server for communication with chromecast, therefore, you can declare an /action endpoint that receives as parameter this signals, and from any frontend technology then you can call this by doing an HTTP call, another way would be sockets, but frankly I prefer HTTP :)

I see the point now. You require some HTTP actions that trigger the different functionalities that already are available for the client. 

>And for your second question, my idea since the beginning was to do a standalone android (and iOS perhaps) app using react native, as a remote controller for the program.

Oh, that's interesting. You want to create a remote app that can be used from Android/iOS based off react. Let me see how to add those HTTP functionalities that you would need. I believe that that should not be too much work. But there is something I have to mention here -- for `forward` and `backward` actions, the thing is working fine when using the `node` backend. That is because the video file is read in full and then streamed so I can basically use `pychromecast` to seek the time I want to. But when using FFmpeg, then the case is different in the sense that when using FFmpeg what mkcrhomecast does is live streaming. Then,  I have to figure out how to relaunch the FFmpeg command to address this. For the moments, I can work with the pause/resume and volume up/ volume down. 

Is react crossplatform? that is -- can that app that you write run on a desktop too?

**Comment by bretanac93** [comment]
Yeah, for desktop we can use [Electron](https://electronjs.org/languages), which basically ports a web app to desktop.

**Comment by ghomem** [comment]
I would be happy to at least be able to control the stream (play, pause, rewind) directly from the computer that streams it. 

As an improvemente we have remoete controle. But I don't think an app is necessary: running an http service on the computer that is streaming, that exposes pause/play/fwd/rewind would be satisfactory.
